secretsmanager/src/base64.cpp

```diff
@@ -19,22 +19,79 @@
 #include <cryptopp/base64.h>
 
 namespace concord::secretsmanager {
-
-std::string base64Enc(const std::vector<uint8_t>& cipher_text) {
+string base64Enc(const vector<uint8_t>& cipher_text) {
+#ifdef USE_CRYPTOPP_RSA
   CryptoPP::Base64Encoder encoder;
   encoder.Put(cipher_text.data(), cipher_text.size());
   encoder.MessageEnd();
   uint64_t output_size = encoder.MaxRetrievable();
-  std::string output(output_size, '0');
+  string output(output_size, '0');
   encoder.Get((unsigned char*)output.data(), output.size());
 
   return output;
+#elif USE_EDDSA_SINGLE_SIGN
+  if (cipher_text.capacity() == 0) {
+    return {};
+  }
+  BIO* b64 = BIO_new(BIO_f_base64());
+  BIO* bio = BIO_new(BIO_s_mem());
+  bio = BIO_push(b64, bio);
+  BIO_write(bio, cipher_text.data(), cipher_text.capacity());
+
+  BUF_MEM* bufferPtr{nullptr};
+  BIO_get_mem_ptr(bio, &bufferPtr);
+  BIO_set_close(bio, BIO_NOCLOSE);
+  BIO_flush(bio);
+  BIO_free_all(bio);
+
+  string encodedMsg;
+  encodedMsg.reserve((*bufferPtr).length);
+
+  for (size_t i{0UL}; i < (*bufferPtr).length; ++i) {
+    encodedMsg += (unsigned char)(*bufferPtr).data[i];
+  }
+  BUF_MEM_free(bufferPtr);
+  return encodedMsg;
+#endif
 }
 
-std::vector<uint8_t> base64Dec(const std::string& input) {
-  std::vector<uint8_t> dec;
+vector<uint8_t> base64Dec(const string& input) {
+#ifdef USE_CRYPTOPP_RSA
+  vector<uint8_t> dec;
   CryptoPP::StringSource ss(input, true, new CryptoPP::Base64Decoder(new CryptoPP::VectorSink(dec)));
+  return dec;
+#elif USE_EDDSA_SINGLE_SIGN
+  if (input.empty()) {
+    return {};
+  }
+  string decodedOutput;
+  decodedOutput.reserve(calcDecodeLength(input.data()));
+
+  BIO* bio = BIO_new_mem_buf(input.data(), -1);
+  BIO* b64 = BIO_new(BIO_f_base64());
```

Comment:
std::unique_ptr<BIO,BIOFreeAll> b64(BIO_new(BIO_f_base64()));
better to wrap this in a smart pointer like above